### DISCUSSION TITLE
Audio recording element

### DISCUSSION THREAD

**Discussion by frankhuurman** [zauberzeug/nicegui#1795] [Ideas / Feature Requests]
Hey team!

I was messing around with some NiceGUI ideas and got audio recording functionality working using the sounddevice PyPi package in combination with a NiceGUI UI.
I'd love to see a native element for this as well and was actually thinking of building the element myself if the team approves of the idea.

I've already checked the contributing readme and thought it would be a perfect way to start contributing to the framework if I can get the dev environment up and running.

I'm unsure whether it's the best cross-platform solution to record audio but I first wanted to get the idea out before comparing other solutions.

Thoughts? :smile:

**Comment by falkoschindler** [comment]
That's a great idea, @frankhuurman! We would definitely appreciate a pull request for such an element.
Actually, there's already a feature request: https://github.com/zauberzeug/nicegui/discussions/569. This could serve as a starting point.
